Lib/axisregistry/__init__.py

```diff
@@ -1,52 +1,458 @@
+from copy import deepcopy
+from fontTools.ttLib import TTFont
+from fontTools.otlLib.builder import buildStatTable
+from fontTools.varLib.instancer.names import _updateUniqueIdNameRecord, NameID
+from fontTools.ttLib.tables._f_v_a_r import NamedInstance
 from pkg_resources import resource_filename
 from google.protobuf import text_format
-
+from collections import OrderedDict
 from axisregistry.axes_pb2 import AxisProto
+from collections import defaultdict
+from itertools import chain
+import logging
+from glob import glob
+import os
 
 try:
     from ._version import version as __version__  # type: ignore
 except ImportError:
     __version__ = "0.0.0+unknown"
 
+log = logging.getLogger(__file__)
+
+# TODO we may have more of these. Please note that some applications may not
+# implement variable font style linking.
+LINKED_VALUES = {
+    "wght": {400.0: 700.0},
+    "ital": {0.0: 1.0},
+}
+
+# Static font styles. The GF api only support the following static font styles
+GF_STATIC_STYLES = OrderedDict(
+    [
+        ("Thin", 100),
+        ("ExtraLight", 200),
+        ("Light", 300),
+        ("Regular", 400),
+        ("Medium", 500),
+        ("SemiBold", 600),
+        ("Bold", 700),
+        ("ExtraBold", 800),
+        ("Black", 900),
+        ("Thin Italic", 100),
+        ("ExtraLight Italic", 200),
+        ("Light Italic", 300),
+        ("Italic", 400),
+        ("Medium Italic", 500),
+        ("SemiBold Italic", 600),
+        ("Bold Italic", 700),
+        ("ExtraBold Italic", 800),
+        ("Black Italic", 900),
+    ]
+)
+
+
+def load_protobuf(klass, path):
+    message = klass()
+    with open(path, "rb") as text_data:
+        text_format.Merge(text_data.read(), message)
+    return message
+
+
+class AxisRegistry:
+    def __init__(self, fp=resource_filename("axisregistry", "data")):
+        axis_fps = [fp for fp in glob(os.path.join(fp, "*.textproto"))]
+        self._data = {}
+        for fp in axis_fps:
+            axis = load_protobuf(AxisProto, fp)
+            self._data[axis.tag] = axis
+
+    def __getitem__(self, k):
+        return self._data[k]
+
+    def __iter__(self):
+        for i in self._data:
+            yield i
+
+    def get_fallback(self, name):
+        for a in self:
+            for fallback in self[a].fallback:
+                if name == fallback.name:
+                    return a, fallback
+        return None, None
+
+    def fallbacks_in_fvar(self, ttFont):
+        res = defaultdict(list)
+        axes_in_font = {
+            a.axisTag: {"min": a.minValue, "max": a.maxValue}
+            for a in ttFont["fvar"].axes
+        }
+        for axis in self:
+            if axis not in axes_in_font:
+                log.warn(f"Axis {axis} not found in GF Axis Registry!")
```

Comment:
Is this the right way around? It doesn't test if the axis from the font is in the registry, it tests if an axis in the registry is in the font. We don't want to warn if a font doesn't have a casual axis, for example.

Maybe you want to iterate over the font's axes, and then grab each axis information from the registry.